www_src/pages/element/element.jsx

```diff
@@ -32,23 +32,34 @@ render(React.createClass({
 
   componentWillMount: function() {
     this.load();
+
+    var saveBeforeSwitch = function() {
+      var goBack = function() {
+        window.Android.goBack();
+      };
+      if (!this.edits) {
+        return goBack();
+      }
+      this.save(goBack);
+    }.bind(this);
+
+    this.props.update({
+      onBackPressed: saveBeforeSwitch
+    });
   },
   componentDidUpdate: function (prevProps) {
     // resume
     if (this.props.isVisible && !prevProps.isVisible) {
       this.load();
     }
-    // pause - if there are edits
-    if (this.edits && !this.props.isVisible && prevProps.isVisible) {
-      this.save();
-    }
   },
   cacheEdits: function (edits) {
     this.edits = edits;
   },
-  save: function () {
+  save: function (postSave) {
```

Comment:
I kind of hold the completely opposite position: calling things `callback` is the worst naming, because it doesn't tell us anything about the variable. I'd be happy to prefix an `on` so that it's obvious that this is a callback function, though, while preserving the immediate information on what it will be used for.